### ISSUE TITLE
The sweep `colon-in-comment` runs `declaration-colon-space-after` under a primary the rule refuses

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#543]
### What minimal example or steps are needed to reproduce the bug?

`scripts/sweeps/colon-in-comment.ts` lists four primary options for `declaration-colon-space-after`, under a JSDoc saying the list is the one `scripts/oracles/options.ts` holds:

```ts
/** Every reader of `raws.between`, under every primary option `scripts/oracles/options.ts` lists for it. */
const configs: Sweep[`configs`] = ([
	…
	[`declaration-colon-space-after`, [`always`, `never`, `always-single-line`, `never-single-line`]],
```

`options.ts` lists three of them, and so does the rule:

```ts
let validOptions = validateOptions(result, ruleName, {
	actual: primary,
	possible: [`always`, `never`, `always-single-line`],
})
```

So Stylelint answers a configuration of `never-single-line` with an `invalidOptionWarning` and the rule never runs.

### What did you expect to happen?

Every configuration a sweep measures under to be one a user can write, so that the rows it keeps are rows the plugin really produces.

### What actually happened?

That configuration is one of the sweep's fifteen, over a corpus of 1 320 texts under three syntaxes, so **3 960 of its 59 400 rows** stand under an option no run of Stylelint can reach. They are not empty rows either: the result of the merged run is still in the store, and **1 628 of the 3 960 hold a warning**.

```shell
$ python3 -c "
import json, os
p = os.path.expanduser('~/.cache/stylelint-stylistic/sweeps/colon-in-comment/8c337f83bf0e90ebe4e0e061.json')
rows = json.load(open(p))
nsl = {k: v for k, v in rows.items() if '|\"never-single-line\"|' in k}
print(len(nsl), sum(1 for v in nsl.values() if v.get('warnings')))
"
3960 1628
```

Those rows are inside the 59 400 runs the commit `5cf5edb` quotes and inside the per-rule counts of moved rows it gives, `630 of declaration-colon-space-after` among them. How many of its 1 829 moved rows stand under the refused option is not measured here: the base side of that comparison is no longer in the store.

### Notes

Found on the branch for #540, by the guard that issue puts back. Until it, the runner of the sweeps and the oracles handed every rule a configuration carrying no `validate` flag, so `validateOptions` accepted every option of every rule and this one ran on whatever branch its checker falls through to. With the flag carried, the rule refuses the option and the rows come back unusable — which is how the drift showed at all.

A probe putting every configuration of every sweep to the runner reports this one and no other: 431 configurations across the eighteen sweeps, one refused.

### Do you have a proposal to fix the bug?

Drop `never-single-line` from the list, which is what the file's own JSDoc already says the list is. What that does to the sweep's finding has to be measured with it, against the counts `5cf5edb` quotes.
